dist/joii.js

```diff
@@ -1176,6 +1207,96 @@
             }
         }
 
+
+        /**
+         * Serializes all serializable properties of an object. Public members are serializable by default.
+         *
+         * @return JSON string containing the serialized class
+         */
+        definition.prototype.Serialize = function ()
```

Comment:
I'm not sure how I would access the private/protected properties from the JOII scope though...

How about just checking if there's a user implementation of the function already, and not overriding if so?